### PR TITLE
Support pre/post actions with `launch_path`

### PR DESCRIPTION
**Pull request by luispadron**
Follow up to #2785 which added `launch_path`. After using this more I realized we can also support `pre`/`post` actions for this attribute.

### CODE REVIEW

File: tools/generators/xcschemes/src/Generator/SchemeInfo.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ struct SchemeInfo: Equatable {
         let scriptText: String
         let action: Action
         let isPreAction: Bool
-        let target: Target
+        let target: Target?
```
Comment (luispadron) [inline tools/generators/xcschemes/src/Generator/SchemeInfo.swift:18]: I don't love this change but I'm not sure how else to represent the execution actions which have no associated build targets (i.e. `launch_path`)

Comment (brentleyjones) [reply tools/generators/xcschemes/src/Generator/SchemeInfo.swift:18]: With your change it makes sense. And technically we could allow in the future execution actions without the macro expansion set, but the current API doesn't allow that.

File: xcodeproj/internal/xcschemes/xcschemes_execution.bzl [hunk 2]
```diff
@@ -317,7 +317,7 @@ def _write_schemes(
             _add_execution_action(
                 action,
                 action_name = action_name,
-                id = target.id,
+                id = getattr(target, "id", None),
```
Comment (brentleyjones) [inline xcodeproj/internal/xcschemes/xcschemes_execution.bzl:320]: I think checking `is_path` in an inline `if else` would be more performant.

Comment (luispadron) [reply xcodeproj/internal/xcschemes/xcschemes_execution.bzl]: I dont have access to `is_path` in here without adding another argument. I moved the `getattr` to be outside the for loop though

Comment (brentleyjones) [reply xcodeproj/internal/xcschemes/xcschemes_execution.bzl]: But `target` is `launch_target`, which has `.is_path`, right?

Comment (luispadron) [reply xcodeproj/internal/xcschemes/xcschemes_execution.bzl]: It wont always be, it looks like this is called for test targets too which wouldn't have `is_path`. But maybe I'm missing something

Comment (brentleyjones) [reply xcodeproj/internal/xcschemes/xcschemes_execution.bzl]: Ahh. Can we change it so we pass in `id` then? We already do an `if` on `is_path`, so we can assign to the variable in both paths there.

Comment (luispadron) [reply xcodeproj/internal/xcschemes/xcschemes_execution.bzl]: Updated!

File: xcodeproj/internal/xcschemes/xcschemes_execution.bzl [hunk 3]
```diff
@@ -317,27 +318,31 @@ def _write_schemes(
...
-                id = target.id,
+                id = target_id,
                 is_pre_action = FALSE_ARG,
                 scheme_name = scheme_name,
             )
 
     # buildifier: disable=uninitialized
     def _add_launch_target(launch_target, *, action_name, scheme_name):
         custom_scheme_args.add(launch_target.is_path)
+        target_id = None
```
Comment (brentleyjones) [inline xcodeproj/internal/xcschemes/xcschemes_execution.bzl:329]: Nit: can you assign this in the `if`, to reduce a double assignment in the `else` path?

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**